File: spring-web-java/src/main/java/spring/web/java/repository/MemberRepository.java
package spring.web.java.repository;

import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.data.jpa.repository.Query;
import spring.web.java.domain.Member;

import java.util.Optional;

public interface MemberRepository extends JpaRepository<Member, Long> {

    @Query("""
            SELECT m
            FROM Member m
            WHERE m.account = :account
            """)
    Optional<Member> findByAccount(String account);
}

File: spring-web-kotlin/src/main/kotlin/spring/web/kotlin/repository/MemberRepository.kt
package spring.web.kotlin.repository

import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.data.jpa.repository.Query
import spring.web.kotlin.domain.Member
import spring.web.kotlin.repository.custom.MemberCustomRepository
import java.util.Optional

interface MemberRepository : JpaRepository<Member, Long>, MemberCustomRepository {

    @Query(
        """
            SELECT m
            FROM Member m
            WHERE m.account = :account
            """
    )
    fun findByAccount(account: String): Optional<Member>
}